packages/rum/src/domain/segmentCollection/segmentCollection.ts

```diff
@@ -43,7 +51,19 @@ export function startSegmentCollection(
   send: (data: Uint8Array, meta: SegmentMeta, rawSegmentSize: number, flushReason?: string) => void
 ) {
   if (!workerSingleton) {
-    workerSingleton = createDeflateWorker()
+    try {
+      workerSingleton = createDeflateWorker()
+    } catch (error) {
+      display.error('Session Replay recording failed to start: an error occurred while creating the Worker:', error)
+      if (includes(error.message, 'Content Security Policy')) {
+        display.error(
+          'Please make sure CSP is correctly configured ' +
+            'https://docs.datadoghq.com/real_user_monitoring/faq/content_security_policy'
+        )
+      }
+      addErrorToMonitoringBatch(error)
```

Comment:
I think we'll want to be warned about CSP issues, cc @alfonsosantacruz

Reply:
what is the reasoning?
we can't do much to solve it

Reply:
that's right, but sometimes we can't access the app in order to find the issue — that said, I'm in to find a better way to help troubleshooting

Reply:
Wouldn't the console error message help the customer to troubleshot this particular issue?

Reply:
for CSP issue, if the goal is to help **us** to troubleshoot the issue, the current strategy will not help much.
For now, we don't receive monitoring logs from customers and if we don't have access to the app, we won't be able to see the console log either.
Another approach could be to also generate a RUM error event like we do for rate limiting of errors and actions.

That being said, I still think that we should not generate a monitoring event for CSP since if at some point we enable monitoring on customers applications, we will be alerted at each time a customer setup the SDK with CSP which IMO is not useful.

Reply:
> Wouldn't the console error message help the customer to troubleshot this particular issue?

I am for keeping the console error but not the monitoring error

Reply:
> I am for keeping the console error but not the monitoring error
It would help customers find out why Session Replay isn't working. Right now they're asking us, and without telemetry all we can do is to see we aren't getting data. CSP settings are mentioned in the docs, and chances are they aren't read.

+1 on keeping the console print

Reply:
We all agree then. I changed the error reporting to report only when it's not a CSP issue.